ui/app/adapters/role.js

```diff
@@ -6,7 +6,17 @@
 // @ts-check
 import { default as ApplicationAdapter, namespace } from './application';
 import classic from 'ember-classic-decorator';
+import { singularize } from 'ember-inflector';
 @classic
 export default class RoleAdapter extends ApplicationAdapter {
   namespace = namespace + '/acl';
+
+  urlForCreateRecord(modelName) {
+    let baseUrl = this.buildURL(modelName);
+    return singularize(baseUrl);
+  }
+
+  urlForDeleteRecord(id) {
+    return this.urlForUpdateRecord(id, 'role');
+  }
```

Comment:
Ember adapters let us define the PUTs and DELETEs and etc. for a given resource; here, we're saying "The API expects list operations at /roles, and individual operations at /role"